Sources/AblyAssetTrackingInternal/AblyWrapper/DefaultAbly.swift

```diff
@@ -420,8 +432,12 @@ extension DefaultAbly: AblyPublisher {
             let data = try geoJson.toJSONString()
             let message = ARTMessage(name: EventName.raw.rawValue, data: data)
             
-            channel.publish([message]) { error in
+            channel.publish([message]) {[weak self] error in
+                guard let self = self else {
+                    return
```

Comment:
Unwrapping `self`, so that I don't need to write `self?` in a couple of places below. Personally I do that if I have to do `self?` more than once after it, but it also introduces some risks for retain cycles in nested closures, so it's not a rule of thumb.